File: application/src/main/java/com/example/app/common/annotation/DistributedLock.java
package com.example.app.common.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import java.util.concurrent.TimeUnit;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface DistributedLock {
    String key();  // 락 키
    TimeUnit timeUnit() default TimeUnit.SECONDS;
    long leaseTime() default 3L;  // 락이 자동으로 해제될 시간 (초)
    long waitTime() default 2L; // 락을 얻기 위해 대기할 최대 시간 (초)
}

File: module-common/src/main/kotlin/com/example/redis/annotations/DistributedLock.kt
package com.example.redis.annotations

import java.util.concurrent.TimeUnit

@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class DistributedLock(
    val lockKey: String,
    val waitTime: Long = 1,
    val leaseTime: Long = 10,
    val timeUnit: TimeUnit = TimeUnit.SECONDS
)
